### ISSUE TITLE
Negative ordinal numbers in submissions table

### ISSUE THREAD

**Issue by dbemke** [getodk/central#641]
#### Problem description
In one of forms in a project submissions aren’t numbered correctly. There are negative numbers in ordinal numbers of submissions. I tried to reproduce it by adding submissions or checking some other forms with many submissions but the URL below is the only example of the issue I could find so I don’t know what’s the factor causing it or how to reproduce it. The issue doesn’t occur e.g. in https://test.getodk.cloud/#/projects/436/forms/all-widgets/submissions
In Firefox I also noticed that the table isn't even.
#### URL of the page
https://test.getodk.cloud/#/projects/385/forms/all-widgets/submissions
#### Steps to reproduce the problem
1. Open the URL with submissions ("URL of the page").
2. Scroll down and check ordinal numbers of submissions
#### Screenshot
![all385](https://github.com/getodk/central/assets/56479916/65d26b64-b811-4635-b14b-21e70a2ceaa2)

#### Expected behavior
Ordinal numbers in submission should start from 1.
#### Central version shown in version.txt
https://test.getodk.cloud/
versions:
9a9ec7c034de07e990c9c408ca768326e2154c96 (v2023.5.1-10-g9a9ec7c)
+c25bc15087ca045496d3bdbc806ca0752b955216 client (v2023.5.0-64-gc25bc150)
+cfe46c97e6e38d8e551ce253ba47d0b378d19352 server (v2023.5.0-56-gcfe46c97)

#### Browser 
Chrome, Firefox

**Comment by matthew-white** [comment]
This is likely the same issue as #608, though the example you identified is different from the two mentioned in that issue. We should have a fix for #608 soon, and once we do, we should check that all the examples have been fixed, both for that issue and for this one. We think the behavior described in this issue can happen when an instance ID is reused, for example, by two submissions in two different projects.

> In Firefox I also noticed that the table isn't even.

This sounds like #607. Does that sound right?

**Comment by dbemke** [comment]
> This sounds like https://github.com/getodk/central/issues/607. Does that sound right?

Yes

**Comment by lognaturel** [comment]
> Can this duplicate instance IDs when I pull a form from one project and push it to a different project?

Yes, that's almost certainly what makes the bug appear. But the problem is the bug, not that you're making it appear! 😊

**Comment by srujner** [comment]
**Tested with Success!**

**Comment by dbemke** [comment]
**Tested with Success!**
